management/actions_test.go

```diff
@@ -53,175 +35,241 @@ func TestActions(t *testing.T) {
-		r, err := m.Action.List()
-		if err != nil {
-			t.Error(err)
-		}
-		t.Logf("%v\n", r)
-	})
+	actualAction, err := m.Action.Read(expectedAction.GetID())
 
-	t.Run("Deploy", func(t *testing.T) {
-		err = ensureActionBuilt(r)
-		if err != nil {
-			t.Fatal(err)
-		}
-		v, err = m.Action.Deploy(r.GetID())
-		if err != nil {
-			t.Fatal(err)
-		}
-		t.Logf("%v\n", v)
-	})
+	assert.NoError(t, err)
+	assert.Equal(t, expectedAction.GetID(), actualAction.GetID())
+}
 
-	t.Run("Update", func(t *testing.T) {
-		id := r.GetID()
+func TestActionManager_Update(t *testing.T) {
+	expectedAction := givenAnAction(t)
+	defer cleanupAction(t, expectedAction.GetID())
 
-		r.ID = nil        // read-only
-		r.UpdatedAt = nil // read-only
-		r.CreatedAt = nil // read-only
-		r.Status = nil    // read-only
-		r.Code = auth0.String("exports.onExecutePostLogin = async (event, api) => { api.user.setUserMetadata('myParam', 'foo'); };")
+	actionID := expectedAction.GetID()
 
-		err = m.Action.Update(id, r)
-		if err != nil {
-			t.Error(err)
-		}
-		t.Logf("%v\n", r)
-	})
+	expectedAction.ID = nil        // Read-Only: Additional properties not allowed.
+	expectedAction.UpdatedAt = nil // Read-Only: Additional properties not allowed.
+	expectedAction.CreatedAt = nil // Read-Only: Additional properties not allowed.
+	expectedAction.Status = nil    // Read-Only: Additional properties not allowed.
 
-	t.Run("DeployAgain", func(t *testing.T) {
-		err = ensureActionBuilt(r)
-		if err != nil {
-			t.Fatal(err)
-		}
-		v, err = m.Action.Deploy(r.GetID())
-		if err != nil {
-			t.Fatal(err)
-		}
-		t.Logf("%v\n", v)
-	})
+	expectedCode := "exports.onExecutePostLogin = async (event, api) => { api.user.setUserMetadata('myParam', 'foo'); };"
+	expectedAction.Code = &expectedCode
 
-	t.Run("Version", func(t *testing.T) {
-		v, err := m.Action.Version(r.GetID(), v.GetID())
-		if err != nil {
-			t.Error(err)
-		}
-		t.Logf("%v\n", v)
-	})
+	err := m.Action.Update(actionID, expectedAction)
 
-	t.Run("Versions", func(t *testing.T) {
-		vl, err = m.Action.Versions(r.GetID())
-		if err != nil {
-			t.Error(err)
-		}
-		t.Logf("%v\n", vl)
-	})
+	assert.NoError(t, err)
+	assert.Equal(t, expectedCode, *expectedAction.Code)
+}
 
-	t.Run("DeployVersion", func(t *testing.T) {
-		v, err = m.Action.DeployVersion(r.GetID(), vl.Versions[0].GetID())
-		if err != nil {
-			t.Fatal(err)
-		}
-		t.Logf("%v\n", v)
-	})
+func TestActionManager_Delete(t *testing.T) {
+	expectedAction := givenAnAction(t)
 
-	t.Run("UpdateBindings", func(t *testing.T) {
-		b := []*ActionBinding{
-			{
-				Ref: &ActionBindingReference{
-					Type:  auth0.String(ActionBindingReferenceByName),
-					Value: r.Name,
-				},
-				DisplayName: auth0.String("My test action Binding"),
+	err := m.Action.Delete(expectedAction.GetID())
+
+	assert.NoError(t, err)
+
+	actualAction, err := m.Action.Read(expectedAction.GetID())
+
+	assert.Empty(t, actualAction)
+	assert.EqualError(t, err, "404 Not Found: That action does not exist.")
```

Comment:
Any way we can assert by the status code (404), so the test doesn't break in the event that the error description changes?

Reply:
Yep, good idea 👍🏻  [954a359](https://github.com/auth0/go-auth0/pull/49/commits/954a3599e0b5ea0f018239b53bd9af5f261bac8f)